src/app/course-manager/duration-input/duration-input.component.ts

```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+
+@Component({
+  selector: 'app-duration-input',
+  templateUrl: './duration-input.component.html',
+  styleUrls: ['./duration-input.component.sass']
+})
+export class DurationInputComponent implements OnInit {
+  @Input() parentForm: FormGroup;
+
+  ngOnInit() {
+    console.log(this.parentForm);
```

Comment:
The same. This component looks like a presentation component. Then you can add `OnPush` strategy to it.